.github/workflows/main.yaml

```diff
@@ -38,29 +40,49 @@ jobs:
       with:
         fetch-depth: 2
 
-    # Skip CI for commits with a message containing the skip-ci string.
-    - name: Fail on skip-ci
-      if: ${{ github.event.after }}
-      run: git log -1 --pretty=format:"%s" ${{ toJSON(github.event.after) }} | grep -v 'skip-ci'
-
     # Explicitly run our scripts with Bash, not PowerShell (GitHub's default).
     - name: Detect current OS
       run: bash ./brink.sh detect_os
 
     - name: Build Python
       run: bash ./pythia build
 
-    - uses: actions/upload-artifact@v2
-      with:
-        name: pythia-${{ matrix.runs-on }}
-        path: dist/**/*.tar.gz
-
     - name: Test Python
       run: bash ./pythia test
 
+    # To use an RSA key with SFTPPlus, install upstream OpenSSH package,
+    # which is more finicky in regards to file permissions.
+    # Beware the commands in this step run under PowerShell.
+    - name: Upload testing package
+      env:
+        BIN_PRIV_KEY: ${{ secrets.SFTPPLUS_BIN_PRIV_KEY }}
+        BIN_HOST_KEY: ${{ secrets.SFTPPLUS_BIN_HOST_KEY }}
+      run: |
+        mkdir -p ~/.ssh/
```

Comment:
It would  be nice to put the "setup" steps in a group... as all this file creation + choco install is not that important.
 
the important part is the call... maybe call it with `-v`

The current publish log end in limbo :)

I would  like to see a confirmation message like:
```
Package was succesfully pushed to server + path.
You can get it from: https://bin.chevah.com/testing/bla
```

In this way, it should  be easier to check the action and go to the result.

Reply:
Agreed. Thanks for the suggestions!

Reply:
Although not sure what you mean by _maybe call it with `-v`_ in this particular section. Call `mkdir -p` with `-v`? It's pretty verbose on Windows anyway:
```
Run mkdir -p ~/.ssh/
52  
53      Directory: C:\Users\runneradmin
54  
55   Mode                 LastWriteTime         Length Name
56  ----                 -------------         ------ ----
57  d----           12/9/2020 11:50 AM                .ssh
58  processed file: .\priv_key
59  Successfully processed 1 files; Failed processing 0 files
60  processed file: .\priv_key
61  Successfully processed 1 files; Failed processing 0 files
```
Same for the other commands… Maybe add `-v` for `mkdir` on non-Windows platforms?

Reply:
Also, constructing the full download link in GitHub actions is not straightforward. It's easy to add the generic link (https://bin.chevah.com:20443/testing/python/), but to show the full link, I think the easiest way would be too add another parameter for the `pythia` script.

Or do you have a better idea?

Reply:
My idea is to have the basic python script export all the env vars it uses for the run / last-run.

In this way, you don't have to construct the batch script inside pythia.

I feel that this can help break the pythia script into smaller pieces and it makes the script easier to read.

Then you can write a python script that does whatever you want in a nice way :1234: 

The verbose suggestion was for the `sftp` command. In the log, I don't see any confirmation from OpenSSH SFTP that the file was uploaded.

Reply:
The `pythia` script does export all the needed variables. However, the resulting env vars would not be available to another process. And not to another step in GitHub Actions.

So I guess you mean writing them down to a file, like we do with `$OS` and `$ARCH` in `BUILD_ENV_VARS`? We could use this file, as only `PYTHON_PACKAGE_VERSION` is missing from there.

I also don't understand what do you mean when suggesting not constructing the batch script from pythia. There, it's done in a single place for all OS'es.  Constructing it with GitHub Actions would mean having the same stuff in a more verbose form in at least 3 different places (for now). Or you meant something else entirely?

As for the pythia script, first of all the batch constructing stuff is not in `pythia.sh`. Also, this is not `chevah_buid` from python-package, `pythia.sh` is 4 times smaller than that. Moving to pythia was done specifically to "break the script into smaller pieces that are easier to read". And I did it for the whole sha-bang, you can compare `functions.sh` too or the scripts from `/src/Python-Windows`, for example. I know there's more that can be done, but the low-hanging fruit has been harvested already.

Also don't get the suggestion about writing a Python script to do it all in a nice way. You mean rewriting pythia in Python? To generate a Python runtime? After barely managing to reproduce python-package's functionality? :-]